test/src/helper/promise.ts

```diff
@@ -17,3 +17,36 @@
 export async function wait(duration: number) {
     await new Promise(resolve => setTimeout(() => resolve(), duration));
 }
+
+export class PromiseExpect {
+    public waiting: { [index: string]: number };
+
+    constructor() {
+        this.waiting = {};
+    }
+
+    public async shouldFulfill<T>(key: string, p: Promise<T>): Promise<T> {
+        if (!this.waiting[key]) {
+            this.waiting[key] = 1;
+        } else {
+            this.waiting[key] += 1;
+        }
+
+        const result = await p;
+
+        this.waiting[key] -= 1;
+        if (this.waiting[key] === 0) {
+            delete this.waiting[key];
+        }
+        return result;
+    }
+
+    public checkFulfil() {
```

Comment:
What does this method?
It just prints log and returns nothing?

Reply:
@sgkim126 
Hyunsik is right. When running the Tendermint tests, there were many test failures caused by connection error. If we could check the reason for the timeout, we could save a lot of time.

Reply:
How about making `checkFulfilled` throw an error if there is a waiting promise?

Reply:
And please use template string instead of string concatenation.